hermes/src/store.rs

```diff
@@ -277,14 +304,41 @@ impl Store {
             })
             .collect::<Result<Vec<_>>>()?;
 
-        let update_data = construct_update_data(messages.iter().collect())?;
+        let update_data = construct_update_data(messages.into_iter().map(|m| m.into()).collect())?;
 
         Ok(PriceFeedsWithUpdateData {
             price_feeds,
-            wormhole_merkle_update_data: update_data,
+            update_data,
         })
     }
 
+    pub async fn get_price_feeds_with_update_data(
+        &self,
+        price_ids: Vec<PriceIdentifier>,
+        request_time: RequestTime,
+    ) -> Result<PriceFeedsWithUpdateData> {
+        match self
+            .get_price_feeds_with_update_data_from_storage(price_ids.clone(), request_time.clone())
+            .await
+        {
+            Ok(price_feeds_with_update_data) => Ok(price_feeds_with_update_data),
+            Err(e) => {
+                if let RequestTime::FirstAfter(publish_time) = request_time {
+                    if let Some(benchmarks_endpoint) = &self.benchmarks_endpoint {
+                        return benchmarks::get_price_feeds_with_update_data_from_benchmarks(
```

Comment:
I'm very wary of the fact our storage abstraction falls back to a remote RPC. I feel like we should be separating out the abstraction but I'm not sure the best way just yet so I think this is fine for this PR, but indicates we might not have the right abstraction as we extend hermes in the future.

Reply:
I will address it a newer PR